Issue #10971: Add example to show how to configure the check with different format for final and non-final pattern variables

mahfouz72 opened the pull request:
closes #10971

mahfouz72 commented:
Github, generate site

mahfouz72 commented:
@rnveach ping

rnveach reviewed (changes requested):
From issue ( https://github.com/checkstyle/checkstyle/issues/10971#issuecomment-2147920381 ) :

> Also your example only showed how to suppress one form of patterns. To validate both, you would have to add 2 instances of PatternVariableName and 2 SuppressionXpathSingleFilter, both with different IDs, to ensure everything is validated.

> previously mentioned documentation update that the local variable checks do not cover patterns, and to go to the pattern version of the check

These don't seem to be done.

mahfouz72 commented:
@rnveach done
- updated example 4 to show how to validate both (final and non final) with different format
- updated doc for `LocalVariableNameCheck` to point to `PatternVariableNameCheck`
- updated doc for `FinalLocalVariableNameCheck` to point to `PatternVariableNameCheck`

mahfouz72 commented:
github, generate site

rnveach commented on src/xdocs/checks/naming/localvariablename.xml.template:
```diff
@@ -13,6 +13,11 @@
         </p>
+        <p>
+          This check does not support pattern variables. Instead, use
+          <a href="https://checkstyle.org/checks/naming/patternvariablename.html#PatternVariableName">
```
links in our documentation must be relative if they are for our own domain.

mahfouz72 replied:
The relative link is not working can you correct me ?

mahfouz72 replied:
@rnveach please help the link is not working now. Can you help me what is wrong?

mahfouz72 replied:
Yeah, It should work now. but still link check plugin is falling is this expected? I think it don't recognize the relative paths and always violate them right?

rnveach replied:
> `+<a href="patternvariablename.html#PatternVariableName">patternvariablename.html#PatternVariableName</a>: doesn't exist.`
`+<a href="patternvariablename.html#PatternVariableName">patternvariablename.html#PatternVariableName</a>: doesn't exist.`

These shouldn't be showing up.

rnveach replied:
@mahfouz72 ping. This is still not done.

rnveach commented on src/xdocs/checks/naming/localfinalvariablename.xml.template:
```diff
@@ -14,6 +14,11 @@
         </p>
+        <p>
+          This check does not support final pattern variables. Instead, use
+          <a href="https://checkstyle.org/checks/naming/patternvariablename.html#PatternVariableName">
```
same.

mahfouz72 commented:
github, generate site

mahfouz72 commented:
github, generate site

rnveach commented on src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheck.java:
```diff
@@ -30,6 +30,11 @@
  * </p>
+ * <p>
+ * This check does not support final pattern variables. Instead, use
+ * <a href="patternvariablename.html#PatternVariableName">
```
Javadoc has to be fully qualified URL. Only xdoc can't be.

mahfouz72 replied:
Yeah, make sense. done

mahfouz72 replied:
xwiki failure is unrelated

mahfouz72 replied:
@rnveach ping

mahfouz72 commented:
github, generate site

mahfouz72 commented:
Rebased
